adapters/Vungle/VungleAdapter/GADMAdapterVungleInterstitial.m

```diff
@@ -25,34 +27,89 @@ - (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)conne
   self = [super init];
   if (self) {
     self.connector = connector;
+    self.adapterAdType = Unknown;
   }
   return self;
 }
 
 - (void)dealloc {
-  [self stopBeingDelegate];
+  //[self stopBeingDelegate];
 }
 
+#pragma mark - GAD Ad Network Protocol Banner Methods (MREC)
+
 - (void)getBannerWithSize:(GADAdSize)adSize {
-  NSError *error = [NSError
-      errorWithDomain:kGADMAdapterVungleErrorDomain
-                 code:0
-             userInfo:@{NSLocalizedDescriptionKey : @"Vungle doesn't support banner ads."}];
-  [_connector adapter:self didFailAd:error];
-}
+  self.adapterAdType = MREC;
 
-- (void)loadInterstitialAd {
-  NSError *error = [[VungleRouter sharedInstance] loadAd:desiredPlacement withDelegate:self];
-  if (error) {
-    [_connector adapter:self didFailAd:error];
+  // Check if given banner size is in MREC.
+  if (!CGSizeEqualToSize(adSize.size, kGADAdSizeMediumRectangle.size)) {
```

Comment:
Please use `IsGADAdSizeValid` to verify ad size as shown below:

```
// An array of supported ad sizes.
NSArray *potentials = @[NSValueFromGADAdSize(kGADAdSizeMediumRectangle)]; 

GADAdSize closestSize = GADClosestValidSizeForAdSizes(adSize, potentials);
if (!IsGADAdSizeValid(closestSize)) {
 // throw error...
}
```

Reply:
@joshuajplagonera  OK, got it, thanks a lot for giving us detailed implementations

Reply:
@joshuajplagonera Thanks, done. 😊